src/program.js

```diff
@@ -63,14 +63,16 @@ export class Program {
     this.yargs.strict();
 
     this.commands = {};
-    this.defaultValues = {};
+    this.mainCommandsList = [];
+    this.options = {};
   }
 
   command(
     name: string, description: string, executor: Function,
     commandOptions: Object = {}
   ): Program {
-    this.defaultValues[name] = setDefaultValues(commandOptions);
+    this.options = Object.assign(this.options, commandOptions);
+    this.mainCommandsList.push(name);
```

Comment:
That would create issues with having the list of options using `Object.keys()` since they won't be the top-level options anymore.